### PR TITLE
Dawn: Support CreatePipeline, ClearColor, ClearDepth, ClearStencil

### PR DESCRIPTION
**Pull request by dneto0**
Right now we only record information that will be used to create
a pipeline.

### CODE REVIEW

File: src/dawn/pipeline_info.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  const ClearColorCommand& GetClearColorValue() const {
+    return clear_color_value_;
+  }
+
+  void SetClearDepthValue(float depth) { clear_depth_value_ = depth; }
+  float GetClearDepthValue() const { return clear_depth_value_; }
+
+  void SetClearStencilValue(float depth) { clear_stencil_value_ = depth; }
+  float GetClearStencilValue() const { return clear_stencil_value_; }
```
Comment (dneto0) [inline src/dawn/pipeline_info.h]: d'oh should be uint32_t

File: src/dawn/pipeline_info.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    clear_color_value_ = value;
+  }
+  const ClearColorCommand& GetClearColorValue() const {
+    return clear_color_value_;
+  }
+
+  void SetClearDepthValue(float depth) { clear_depth_value_ = depth; }
+  float GetClearDepthValue() const { return clear_depth_value_; }
+
+  void SetClearStencilValue(float depth) { clear_stencil_value_ = depth; }
```
Comment (dj2) [inline src/dawn/pipeline_info.h]: Missed one, should be uint32_t stencil not float depth

File: src/dawn/pipeline_info_test.cc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+#include <cstdint>
+#include "dawn/dawncpp.h"
+#include "gmock/gmock.h"
+
+namespace amber {
+namespace dawn {
+
+namespace {
+
+using ::testing::Eq;
```
Comment (dj2) [inline src/dawn/pipeline_info_test.cc]: What is this for? You never call Eq so this using shoudn't be needed?

File: src/dawn/pipeline_info_test.cc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+TEST_F(DawnRenderPipelineInfoTest, SetClearColor) {
+  RenderPipelineInfo rpi;
+  ClearColorCommand ccc;
+  ccc.SetR(1.0f);
+  ccc.SetG(2.0f);
+  ccc.SetB(3.0f);
+  ccc.SetA(-4.0f);
+  rpi.SetClearColorValue(ccc);
+  const ClearColorCommand& color = rpi.GetClearColorValue();
+  EXPECT_EQ(1.0f, color.GetR());
```
Comment (dj2) [inline src/dawn/pipeline_info_test.cc]: EXPECT_FLOAT_EQ for floats

**Pull request merged**